### PR TITLE
[AIRFLOW-4737] Increase and document celery queue name limit

### PR DESCRIPTION
**Pull request by villasv**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4737
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

As described in the relevant issue and commit body, 50 characters is too conservative, as 100 might as well be. SQS names can be 80 characters long, which served as anchor for picking the next multiple of 50 that would suffice.

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

This change should be transparent and have no effect on existing features.

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

**Comment by feng-tao** [comment]
I think it needs a db migration script

**Comment by villasv** [comment]
I guess I was sleep drunk, I forgot about a migration for that. On it.

File: docs/howto/executor/use-celery.rst [hunk 1]
```diff
@@ -69,3 +69,4 @@ Some caveats:
 - Make sure to use a database backed result backend
 - Make sure to set a visibility timeout in [celery_broker_transport_options] that exceeds the ETA of your longest running task
 - Tasks can consume resources. Make sure your worker has enough resources to run `worker_concurrency` tasks
+- Queue names are limited to 100 characters, but each broker backend might have its own restrictions
```
Comment (ashb) [inline docs/howto/executor/use-celery.rst]: How about 256 chars instead?

Comment (villasv) [reply docs/howto/executor/use-celery.rst]: Sure, will do.

**Comment by villasv** [comment]
BTW @ashb Pylint is failing but the file structure is automatically generated :-/
Should I add a docstring and sort that file name out?

**Comment by ashb** [comment]
@BasPH Pylint issues here :( https://travis-ci.org/apache/airflow/jobs/542703778

```
airflow/migrations/versions/004c1210f153_increase_queue_name_size_limit.py:1:0: C0103: Module name "004c1210f153_increase_queue_name_size_limit" doesn't conform to snake_case naming style (invalid-name)
airflow/migrations/versions/004c1210f153_increase_queue_name_size_limit.py:37:0: C0111: Missing function docstring (missing-docstring)
airflow/migrations/versions/004c1210f153_increase_queue_name_size_limit.py:43:0: C0111: Missing function docstring (missing-docstring)
```

I think we should probably exclude migrations from py-lint checking. WDYT?

**Comment by BasPH** [comment]
I suggest to widen the restrictions of the module name. Opened a PR (https://github.com/apache/airflow/pull/5391) to allow module names to start with 0-9 and now have max 60 chars, this is green on all our airflow/migration module names. @ashb WDYT?

@villasv For the missing-docstring messages I suggest to use the Python docstring format so that Pylint will be happy:
```python
"""use batch_alter_table to support SQLite workaround"""
```

**Comment by villasv** [comment]
Ehhh, I'll do it. But I find enforcing docstrings like that for migrations pretty superfluous. IMO it's better to just disable (or use a different config for) pylint on migrations.

**Comment by potiuk** [comment]
@villasv -> it's lesser of two evils. Currently pylint does not have way to decide which modules should have docstrings and which should not :(. We have an open issue for that https://issues.apache.org/jira/browse/AIRFLOW-4751 - which we might address by PR to pylint, but for now we think it's a small inconvenience and having docstrings for most modules makes sense.

**Comment by BasPH** [comment]
Few things to consider here: migrations are also part of the Airflow codebase, so for consistency sake we like to apply the same conventions.

Every function performs some sort of logic, so it's convenient for other people and even your future self to explain _what_ the function does, even if it's a simple one-liner. Typically business logic code is very hard to debug and understand - because others don't have the context you had while writing the code.

IMO this migration is a perfect example of this - the number of chars is raised. If it ever turns out to be an issue and somebody else would look at the code, they don't see any explanation in the code stating why the number was raised and why 256 was chosen. Only after reading your JIRA issue do I understand why these choices were made, which are perfectly fine. Unfortunately, not everybody explains the what and why in their Git commit/PR/JIRA issue.

Reading the Airflow code base, there are a lot of such examples of undocumented logic, so now that we are in the process of adding Pylint I prefer not to lower the boundaries just because it's inconvenient. If you'd write a DAG at your work and you're the only person maintaining the thing, I wouldn't really be bothered, but Airflow is software used and developed by many people around the world so I strongly prefer to have consistently documented code.

Sorry for the wall of text, hope you understand 😃

**Comment by BasPH** [comment]
Btw https://github.com/apache/airflow/pull/5391 has been merged, so if you rebase on master all should be good now.

**Comment by villasv** [comment]
I'm sorry if it seemed like I was considering it inconvenient and a bad idea, it's not. It's negligible effort, but I do think it's also negligible value.

While migrations are indeed part of the codebase, they're definitely not in the scope of standard software development standards. You said you didn't want to lower the boundaries, but that's sort of what happened: very long modules names starting with numbers are now allowed inside _all_ the codebase, despite this being acceptable only inside migrations. That's why I said migrations should use a different set of pylint configurations (or none at all, but the first is better). 

I completely agree with all your motives, despite this docstring in particular being totally unrelated to the issue itself and IMO better as a line comment like it was before. But it's OK. I think git/JIRA is the right place to document the value decision anyway. 

I see the doscstring currently being merged is not there because it's a good docstring, it's there because I need pylint to pass. This means pylint needs adjustments to become more useful.

Will rebase and push.

**Comment by villasv** [comment]
> I see the doscstring currently being merged is not there because it's a good docstring, it's there because I need pylint to pass. This means pylint needs adjustments to become more useful.

I guess this is also related to how much effort I put. Let me reformulate the docs in this migration.

**Comment by villasv** [comment]
Alright, there it is. I think this docstring is now more useful and actually related to the method itself and not the implementation detail.

I still think module names should be kept strict in the main codebase, but if this is the kind of dosctring that gets enforced, they can be helpful for future maintainers indeed.

**Review by ashb — APPROVED** [review]

**Pull request merged**